File: 1433-check-if-a-string-can-break-another-string/1433-check-if-a-string-can-break-another-string.cpp
class Solution {
public:
bool get_X_greater(string s1, string s2, int l){
        for(int i=l; i<s1.length(); i++){
            if(s2[i] > s1[i]) return false;
        }
        return true;
    }
    
    bool get_Y_greater(string s1, string s2, int l){
        for(int i=l; i<s1.length(); i++){
            if(s1[i] > s2[i]) return false;
        }
        return true;
    }
    
    bool checkIfCanBreak(string s1, string s2) {
        sort(s1.begin(), s1.end());
        sort(s2.begin(), s2.end());
        bool flag;
        for(int i=0; i<s1.length(); i++){
            if(s1[i] != s2[i]){
                if(s1[i] > s2[i]){
                    flag = get_X_greater(s1, s2, i);
                }
                else{
                    flag = get_Y_greater(s1, s2, i);
                }
                break;
            }
        }
        return flag;
    }
};

File: 1433-check-if-a-string-can-break-another-string/1433-check-if-a-string-can-break-another-string.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
function get_grt(s1, s2,l){
        for(var i=l; i<s1.length; i++){
            if(s2[i] > s1[i]) return false;
        }
        return true;
    }

    
var checkIfCanBreak = function(s1, s2) {
    s3=s1.split("").sort().reverse();
    s4=s2.split("").sort().reverse();
    var cmp;
    var i=0;
    var flag;
    for(;i<s3.length;i++)
    {
        if(s3[i]!==s4[i]){
        if(s3[i]>s4[i]){
           flag = get_grt(s3,s4,i);
        }
        else{
            flag = get_grt(s4,s3,i);
        }
            break;
        }
    }
    return flag;
};